### ISSUE TITLE
Drupal 10 readiness

### ISSUE THREAD

**Issue by divya-intelli** [apigee/apigee-devportal-kickstart-drupal#614]
## Is your feature request related to a problem? Please describe
Some of the themes and modules are deprecated in Drupal 10 https://www.drupal.org/docs/core-modules-and-themes/deprecated-and-obsolete , update the modules and themes to be compatible with Drupal 10.

**Comment by hollyw2025** [comment]
Hi, we are waiting for this module to upgrade to D10. 

we have "apigee/apigee_devportal_kickstart": "2.1.2" as required module with our project, which is not D10 compatible at this moment. I tried to upgraded locally, here is what I did

1. upgraded apigee_devportal_kickstart/composer.json to use a couple of modules which are D10 compatible 
"require": {
    "cweagans/composer-patches": "^1.6.5",
    "drupal/admin_toolbar": "^3.0",
    "drupal/adminimal_admin_toolbar": "^1.9",
    "drupal/apigee_api_catalog": "^3",
    "drupal/apigee_edge": "^1.18 || ^2.0.0",
    "drupal/better_exposed_filters": "^6.0",
    "drupal/jquery_ui_datepicker": "^1.2",
    "drupal/default_content": "^2.0@alpha",
    "drupal/email_registration": "^1.0@RC",
    "drupal/fontawesome": "^2.12",
    "drupal/paragraphs": "^1.6",
    "drupal/pathauto": "^1.6",
    "drupal/radix": "^5.0",
    "drupal/components": "^3.0@beta"
  }
  
  2, run composer update under this folder of this module, and I run into a lot of error 
  
  Applying patches for drupal/core
    https://www.drupal.org/files/issues/2018-12-17/core-undefined-index-summary_attributes-2998194-9.patch (Fix issue with #summary_details introduced in Drupal 8.6.x)
   Could not apply patch! Skipping. The error was: The process "patch '-p1' --no-backup-if-mismatch -d '/Users/hollywang/Projects/githubportal/release31/web/profiles/contrib/apigee_devportal_kickstart/vendor/drupal/core' < '/var/folders/ny/c0wz9f9n6qj51g60vg88mtn80000gn/T/64adb213ab846.patch'" exceeded the timeout of 300 seconds.
    https://www.drupal.org/files/issues/2021-04-23/validate-reference-constraint-3210319-3.patch (Fix Validation issue for entity reference field on updating teams #470)
   Could not apply patch! Skipping. The error was: The process "patch '-p1' --no-backup-if-mismatch -d '/Users/hollywang/Projects/githubportal/release31/web/profiles/contrib/apigee_devportal_kickstart/vendor/drupal/core' < '/var/folders/ny/c0wz9f9n6qj51g60vg88mtn80000gn/T/64adb340c569a.patch'" exceeded the timeout of 300 seconds.

  - Applying patches for drupal/default_content
    https://www.drupal.org/files/issues/2020-11-06/default_content-existing-entities-2698425-1-x.patch (Do not import existing entities)
   Could not apply patch! Skipping. The error was: The process "patch '-p0' --no-backup-if-mismatch -d '/Users/hollywang/Projects/githubportal/release31/web/profiles/contrib/apigee_devportal_kickstart/vendor/drupal/default_content' < '/var/folders/ny/c0wz9f9n6qj51g60vg88mtn80000gn/T/64adb46d00df7.patch'" exceeded the timeout of 300 seconds.

please advice

Thank you so much

Holly

**Comment by divya-intelli** [comment]
Hi @HollyWangUCLA ,

We are working on support of Drupal 10. Changes for Drupal 10 will be merged in drupal10-prep branch. 
Once the testing is done we will release a new version which will be compatible with Drupal 10.

**Comment by hollyw2025** [comment]
cool, thank you so much

**Comment by askmahesh1** [comment]
Do you have a timeline on when the Drupal10 compatible version will be available . Our Developer Portal project is on hold as we do not want to roll out with Drupal 9 which has a EOL for Security Support in Nov '23 

Thanks

**Comment by divya-intelli** [comment]
Hi @askmahesh1 , @HollyWangUCLA 
We have released new version of Developer Portal which is Drupal 10 compatible.
Note : Before upgrading from Drupal 9 to Drupal 10 , refer [document](https://www.drupal.org/docs/contributed-modules/apigee-developer-portal-kickstart/upgrade-from-drupal-9-to-drupal-10).
